service/src/main/java/bio/terra/user/app/controller/ProfileApiController.java

```diff
@@ -19,32 +21,44 @@ public class ProfileApiController implements ProfileApi {
   private final HttpServletRequest servletRequest;
   private final ProfileService profileService;
   private final ProfileSamUserFactory userFactory;
+  private final SamService samService;
 
   @Autowired
   public ProfileApiController(
       HttpServletRequest servletRequest,
       ProfileService profileService,
-      ProfileSamUserFactory userFactory) {
+      ProfileSamUserFactory userFactory,
+      SamService samService) {
     this.servletRequest = servletRequest;
     this.profileService = profileService;
     this.userFactory = userFactory;
+    this.samService = samService;
   }
 
   @Override
-  public ResponseEntity<AnyObject> setUserProfile(AnyObject body, String path) {
-    profileService.setProperty(getUser(), parsePath(path), body.getValue());
+  public ResponseEntity<AnyObject> setUserProfile(AnyObject body, String path, String userEmail) {
```

Comment:
not strictly related to this PR, but i think we should always check if the user is a sam user for all the endpoint?

Reply:
Agree, though I think this is already implicitly happening with the current endpoints since they either ask for the current SamUser or attempt to make an admin request where they ofc. have to be a user

Reply:
ah you are right